pkg/controllers/vdb/replicatedupgrade_reconciler.go

```diff
@@ -221,16 +452,61 @@ func (r *ReplicatedUpgradeReconciler) assignSubclustersToReplicaGroupACallback()
 	return annotatedAtLeastOnce, nil
 }
 
+// moveReplicaGroupBSubclusterToSandbox will move all subclusters attached to
+// replica group B into the sandbox.
+func (r *ReplicatedUpgradeReconciler) moveReplicaGroupBSubclusterToSandbox() (bool, error) {
+	oldImage, found := r.Manager.fetchOldImage()
+	if !found {
+		return false, errors.New("Could not find old image")
+	}
+
+	scNames := r.getSubclustersForReplicaGroup(vmeta.ReplicaGroupBValue)
+	if len(scNames) == 0 {
+		return false, errors.New("cound not find any subclusters for replica group B")
+	}
+
+	sandboxName, err := r.getNewSandboxName(preferredSandboxName)
+	if err != nil {
+		return false, fmt.Errorf("failed to generate a unique sandbox name: %w", err)
+	}
+	sandbox := vapi.Sandbox{
+		Name:  sandboxName,
+		Image: oldImage,
+	}
+	for _, nm := range scNames {
+		sandbox.Subclusters = append(sandbox.Subclusters, vapi.SubclusterName{Name: nm})
+	}
+	r.VDB.Annotations[vmeta.ReplicatedUpgradeSandboxAnnotation] = sandboxName
+	r.VDB.Spec.Sandboxes = append(r.VDB.Spec.Sandboxes, sandbox)
```

Comment:
I guess this is vdb is updated elsewhere. Can you add a comment about why we do this here and when/where this will be used?

Reply:
This is used as a callback function `updateVDBWithRetry` to do the actual update.